test/gateway/eip20_gateway/progress_stake_with_proof.js

```diff
@@ -542,4 +546,76 @@ contract('EIP20Gateway.progressStakeWithProof()', function (accounts) {
 
   });
 
+  it('should return penalty to staker when the message status in source is ' +
+    'declared revocation and in the target is progressed', async function () {
+
+    let stakeVault = await gateway.stakeVault.call();
+    let caller = accounts[0];
+    let staker = stakeData.staker;
+
+    await gateway.setOutboxStatus(
+      stakeData.messageHash,
+      MessageStatusEnum.DeclaredRevocation,
+    );
+
+    // fund Gateway with enough tokens that it can return penalty
+    await baseToken.transfer(gateway.address, penaltyAmount, { from: accounts[0] });
+
+    let callerInitialBaseTokenBalance = await baseToken.balanceOf(caller);
+    let stakerInitialBaseTokenBalance = await baseToken.balanceOf(staker);
+    let gatewayInitialTokenBalance = await mockToken.balanceOf(gateway.address);
```

Comment:
As we already have a separate test case to assert bounty and mocktoken balance, we can exclude that from this test.

Reply:
I think the additional asserts are still valuable to check that returning the penalty doesn't have any side effects on the bounty (e.g. the caller doesn't receive the bounty instead of the staker) or Gateway balance.